Automattic/wp-calypso
# Store: Add "product weight" and "product value" fields to the Customs form

Note: This should be tested using the `update/add-customs-weight-and-value-to-meta` branch on `woocommerce-services`.

I hope the commits are more-or-less self-explanatory because you non-AFK Hydras will be stuck improving any annoyance. I've done some testing with a product with no weight and $0 price and it worked fine, but this could use some more testing.

This PR adds a `Declared Value` and `Weight` fields to the items of the Customs form. Like every other field in the flow, it has validation that only triggers after the user has interacted with the field or the step has been submitted.

The things I would improve if I had more time are:
* Styles, definitely. This is a blocker, but I don't know how to improve it (hoping that pinging designers will help)
* Using the `ignoreTariffNumberValidation` pattern, copy&pasting it exactly the same way for the weight and value fields. Maybe we should do the `every field is valid until interacted with` logic in another cleaner way.
* The values for `value` and `weight` can be `string` or `number` depending on their source. This should be normalized at the reducer level, instead of letting the components just accept any type of value.

@kellychoffman The mobile styles look fine, but the desktop styles produce this monstrosity:
<img width="547" alt="screen shot 2018-07-31 at 21 17 20" src="https://user-images.githubusercontent.com/1715800/43483109-f3a9d17a-950a-11e8-850b-965602da7bd1.png">

I didn't even try to style it because there's no way those 5 columns physically fit into the modal. Do you have any ideas on how to structure this?

**To test:**
- Create or edit a product so it has no weight and/or no price (`$0`)
- Create an order with that product and with an international destination. Note that you won't be able to use `USPS` with a weightless product, so you should use Free Shipping or Flat Rate. Also, you won't be able to use the `Checkout` for a `$0` order so you should add another item if you're testing the `$0` product.
- Go print a label for that order.
- Make the screen narrow so you get the mobile styles for the `Customs` form. That's the only useable UI for now.
- See that in the Customs form, the item without weight/price has those fields blank or zero.
- Check you can edit those fields, and validation works (no non-numeric strings, no negative or zero values allowed).
- Refresh the page, go to the Customs form again and submit it. Check that the errors didn't show before, but they show after you clicked the `Submit` button.
- After putting valid values for `weight` and `value`, submit the step so rates can be fetched.
- Refresh the page, go to the Customs form again, check that the values you entered were persisted.
- Check that the real product weight was updated (you can go to WP-Admin for that).
